boltons/iterutils.py

```diff
@@ -177,6 +178,100 @@ def split_iter(src, sep=None, maxsplit=None):
     return
 
 
+def lstrip(iterable, strip_value=None):
+    """Strips values from the beginning of an iterable. Stripped items will
+    match the value of the argument strip_value. Functionality is analigous
+    to that of the method str.lstrip. Returns a list.
+
+    >>> lstrip(['Foo', 'Bar', 'Bam'], 'Foo')
+    ['Bar', 'Bam']
+
+    """
+    return list(lstrip_iter(iterable, strip_value))
```

Comment:
One reason I was hoping to get @kurtbrose in here was to ask whether he would expect lstrip to dynamically handle the string case. I.e., autojoin things back together as "abc".lstrip('a') returns a string, not a list of `['b', 'c']`.

Reply:
I wondered about that too but decided against that because it would be a special case to be used for strings only. These changes were meant to generically handle iterables, a class of value that strings happen to be part of. Besides, if this version of the function is problematic for a developer who is working with strings, there are always the string based methods.